File: app/src/main/java/app/deadmc/devnetworktool/helpers/SystemHelper.java
package app.deadmc.devnetworktool.helpers;

import java.io.BufferedReader;
import java.io.InputStreamReader;

/**
 * Created by Feren on 08.11.2016.
 */
public class SystemHelper {

    public static String executeCmd(String cmd, boolean sudo){
        try {

            Process p;
            if(!sudo)
                p= Runtime.getRuntime().exec(cmd);
            else{
                p= Runtime.getRuntime().exec(new String[]{"su", "-c", cmd});
            }
            BufferedReader stdInput = new BufferedReader(new InputStreamReader(p.getInputStream()));

            String s;
            String res = "";
            while ((s = stdInput.readLine()) != null) {
                res += s;
            }
            p.destroy();
            return res;
        } catch (Exception e) {
            e.printStackTrace();
        }
        return "";

    }

    public static String getPing(String url) {
        String pingString = SystemHelper.executeCmd("ping -c 1 -w 1 "+url, false);
        return pingString;
    }


}

File: app/src/main/java/app/deadmc/devnetworktool/helpers/SystemHelper.kt
package app.deadmc.devnetworktool.helpers

import android.util.Log
import app.deadmc.devnetworktool.shared_preferences.DevPreferences
import kotlinx.coroutines.experimental.Deferred
import kotlinx.coroutines.experimental.async
import kotlinx.coroutines.experimental.delay
import java.io.BufferedReader
import java.io.InputStreamReader

fun executeCmd(cmd: String, sudo: Boolean): String {
    safe {
        val p: Process
        if (!sudo)
            p = Runtime.getRuntime().exec(cmd)
        else {
            p = Runtime.getRuntime().exec(arrayOf("su", "-c", cmd))
        }
        val stdInput = BufferedReader(InputStreamReader(p.inputStream))

        var s: String? = ""
        var res = ""
        while (s != null) {
            s = stdInput.readLine()
            if (s != null) res += s
        }
        p.destroy()
        return res
    }

    return ""

}

fun getPing(url: String): Deferred<String> {
    return async {
        Log.e("getPing","before delay")
        delay(DevPreferences.pingDelay.toLong())
        Log.e("getPing","after delay")
        executeCmd("ping -c 1 -w 1 " + url, false)
    }
}

